src/mallocTracer.cpp

```diff
@@ -154,6 +121,11 @@ void MallocTracer::patchLibraries() {
     while (_patched_libs < native_lib_count) {
         CodeCache* cc = (*native_libs)[_patched_libs++];
 
+        if (cc->contains((const void*)MallocTracer::initialize)) {
+            // Let libasyncProfiler always use original allocation methods
```

Comment:
Async-profiler will not be able to trace its own allocations. That's unfortunate, although acceptable.